ui/pages/send/send-content/send-content.component.js

```diff
@@ -77,11 +121,83 @@ export default class SendContent extends Component {
       recipient.warning === 'knownAddressRecipient';
     const hideAddContactDialog = recipient.warning === 'loading';
 
+    let title;
+    if (
+      draftTransaction?.asset.details?.standard === ERC721 ||
+      draftTransaction?.asset.details?.standard === ERC1155
+    ) {
+      title = draftTransaction?.asset.details?.name;
+    } else if (draftTransaction?.asset.details?.standard === ERC20) {
+      title = `${hexWEIToDecETH(draftTransaction.amount.value)} ${
+        draftTransaction?.asset.details?.symbol
+      }`;
+    }
+
+    const ethTransactionTotalMaxAmount = Number(
+      hexWEIToDecETH(hexMaximumTransactionFee),
+    );
+
+    const primaryTotalTextOverrideMaxAmount = `${title} + ${ethTransactionTotalMaxAmount} ${nativeCurrency}`;
+
+    const detailText = (
+      <Box
+        height={BLOCK_SIZES.MAX}
+        display={DISPLAY.FLEX}
+        flexDirection={FLEX_DIRECTION.COLUMN}
+        className="gas-display__total-value"
+      >
+        <LoadingHeartBeat estimateUsed={draftTransaction?.userFeeLevel} />
+        <UserPreferencedCurrencyDisplay
+          type={SECONDARY}
+          key="total-detail-text"
+          value={hexTransactionTotal}
+          hideLabel={Boolean(useNativeCurrencyAsPrimaryCurrency)}
+        />
+      </Box>
+    );
+
+    let detailTotal, maxAmount;
+
+    if (draftTransaction?.asset.type === 'NATIVE') {
+      detailTotal = (
+        <Box
+          height={BLOCK_SIZES.MAX}
+          display={DISPLAY.FLEX}
+          flexDirection={FLEX_DIRECTION.COLUMN}
+          className="gas-display__total-value"
+        >
+          <LoadingHeartBeat estimateUsed={draftTransaction?.userFeeLevel} />
+          <UserPreferencedCurrencyDisplay
+            type={PRIMARY}
+            key="total-detail-value"
+            value={hexTransactionTotal}
+            hideLabel={!useNativeCurrencyAsPrimaryCurrency}
+          />
+        </Box>
+      );
+      maxAmount = (
+        <UserPreferencedCurrencyDisplay
+          type={PRIMARY}
+          key="total-max-amount"
+          value={addHexes(
+            draftTransaction.amount.value,
+            hexMaximumTransactionFee,
+          )}
+          hideLabel={!useNativeCurrencyAsPrimaryCurrency}
+        />
+      );
+    } else if (useNativeCurrencyAsPrimaryCurrency) {
+      detailTotal = primaryTotalTextOverrideMaxAmount;
+      maxAmount = primaryTotalTextOverrideMaxAmount;
+    } else {
+      detailTotal = undefined;
+      maxAmount = undefined;
+    }
```

Comment:
This else statement is unnecessary, they will be undefined if neither case hits.

Reply:
@brad-decker Done [here](https://github.com/MetaMask/metamask-extension/pull/14634/commits/c64948a3c0494308f824cdc443a9b04b010b9054).